pkg/pillar/pubsub/publish.go

```diff
@@ -151,6 +151,17 @@ func (pub *PublicationImpl) GetAll() map[string]interface{} {
 	return result
 }
 
+// Iterate - performs some callback function on all items
+func (pub *PublicationImpl) Iterate(function CallBackFunction) {
+	items := pub.GetAll()
```

Comment:
Hmm.. Am wondering if we should use GetAll ?? Or can we directly iterate on the KeyMap?
That is more efficient, but using GetAll is Safer - if a PubSub entry is deleted in the function etc.

Lets not use GetAll in the Iterate. You can use the KeyMap.Range function..